sklearn/ensemble/_iforest.py

```diff
@@ -327,6 +327,18 @@ def fit(self, X, y=None, sample_weight=None):
             check_input=False,
         )
 
+        self._average_path_length_max_samples = _average_path_length(
+            [self._max_samples]
+        )
+
+        self._average_path_length_per_tree = []
+        self._decision_path_lenghts = []
```

Comment:
Hi! I'm just checking out your great work, and spotted a small typo for the variable name on this line (also used later):

`_decision_path_lenghts ` -> `_decision_path_lengths`